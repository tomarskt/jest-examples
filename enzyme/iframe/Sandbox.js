import React, { Component } from "react";
import { GelXIframe } from "lerna-test-library";

import DisplayLoader from "./displayloader";

class Sandbox extends Component {
  render() {
    const { url, postMessage } = this.props;
    return (
      <GelXIframe
        displayLoader={<DisplayLoader />}
        url={url}
        postMessage={postMessage}
      />
    );
  }
}

export default Sandbox;
