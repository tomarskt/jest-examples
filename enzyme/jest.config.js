module.exports = {
    "verbose": false,
    "bail": true,
    "testEnvironment": "jsdom",
    "setupFiles": ["<rootDir>/setup/shim.js", "<rootDir>/setup/index.js"],
    "snapshotSerializers": ["enzyme-to-json/serializer"]
}