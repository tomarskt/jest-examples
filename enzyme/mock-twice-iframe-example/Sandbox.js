import React, { Fragment, Component } from "react";
import { GelXIframe } from "lerna-test-library";

import DisplayLoader from "./displayloader";
import { dispatch } from "./store";

class Sandbox extends Component {
  postMessage = e => {
    const {
      data: { value }
    } = e;
    console.warn(
      "********  you need to see me in your unit test  ************"
    );
    dispatch({ value: value + 1 });
  };
  render() {
    const { url, postMessage } = this.props;
    return (
      <Fragment>
        <DisplayLoader />
        <GelXIframe
          displayLoader={<DisplayLoader />}
          url={url}
          postMessage={this.postMessage}
        />
      </Fragment>
    );
  }
}

export default Sandbox;
