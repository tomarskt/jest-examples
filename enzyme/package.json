{
  "scripts": {
    "test": "jest",
    "debug": "node --inspect-brk node_modules/.bin/jest  --runInBand  "
  },
  "dependencies": {
    "react": "^16.4.0",
    "react-dom": "^16.4.0",
    "prop-types": "^15.6.2"
  },
  "devDependencies": {
    "babel-plugin-transform-async-to-generator": "^6.24.1",
    "babel-plugin-transform-class-properties": "^6.24.1",
    "babel-plugin-transform-object-rest-spread": "^6.26.0",
    "babel-polyfill": "^6.26.0",
    "babel-preset-es2015": "*",
    "babel-preset-react": "*",
    "enzyme": "^3.3.0",
    "enzyme-adapter-react-16": "^1.0.1",
    "enzyme-to-json": "^3.3.4",
    "jest": "^23.4.1",
    "react-router-dom": "^4.3.1",
    "react-test-renderer": "^16.4.0",
    "redux-form": "^7.3.0",
    "lerna-test-library": "^1.2.1"
  }
}
