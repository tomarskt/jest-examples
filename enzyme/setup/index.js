import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
// to supress all warnings, uncomment below line;
// no need to install jsdom or setup jsom, as it's part of jest and 
// jest default to jsdom in testEnvironmenet settins. see package.json jest section
// process.env.NODE_ENV = 'production' 


Enzyme.configure({ adapter: new Adapter() });

