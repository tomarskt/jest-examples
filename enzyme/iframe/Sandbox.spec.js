import React from 'react';
import {shallow, mount} from 'enzyme';

import Sandbox from './Sandbox';

jest.mock('./displayloader', ()=> 'MockDisplayLoader') ; //defualt export mock

describe('iframe',()=>{
    it('should render' , ()=>{
        const shallowed = shallow(<Sandbox url='http://unit-test.com' />);
        expect(shallowed).toMatchSnapshot();
    })
    it('should handle postMessage',()=>{


    })
});
