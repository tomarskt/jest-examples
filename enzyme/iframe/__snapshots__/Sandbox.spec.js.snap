// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`iframe should deep render correctly 1`] = `
<Sandbox
  url="http://unit-test.com"
>
  <GelXIframe
    displayLoader={<MockDisplayLoader />}
    postMessage={[Function]}
    url="http://unit-test.com"
  >
    <div>
      mocked ver :iframe source =
      http://unit-test.com
    </div>
  </GelXIframe>
</Sandbox>
`;

exports[`iframe should shallow render correctly  1`] = `
<GelXIframe
  displayLoader={<MockDisplayLoader />}
  postMessage={[Function]}
  url="http://unit-test.com"
/>
`;
